src/Storages/System/StorageSystemDDLWorkerQueue.cpp

```diff
@@ -221,41 +221,46 @@ void StorageSystemDDLWorkerQueue::fillData(MutableColumns & res_columns, Context
     zkutil::ZooKeeperPtr zookeeper = ddl_worker.getAndSetZooKeeper();
     Strings ddl_task_paths = zookeeper->getChildren(ddl_zookeeper_path);
 
-    GetResponseFutures ddl_task_futures;
-    ListResponseFutures active_nodes_futures;
-    ListResponseFutures finished_nodes_futures;
+
+    std::vector<std::string> ddl_task_full_paths;
+    ddl_task_full_paths.reserve(ddl_task_paths.size());
+    std::vector<std::string> ddl_task_status_paths;
+    ddl_task_status_paths.reserve(ddl_task_paths.size() * 2); // for active and finished
 
     for (const auto & task_path : ddl_task_paths)
     {
-        ddl_task_futures.push_back(zookeeper->asyncTryGet(ddl_zookeeper_path / task_path));
+        ddl_task_full_paths.push_back(ddl_zookeeper_path / task_path);
         /// List status dirs. Active host may become finished, so we list active first.
-        active_nodes_futures.push_back(zookeeper->asyncTryGetChildrenNoThrow(ddl_zookeeper_path / task_path / "active"));
-        finished_nodes_futures.push_back(zookeeper->asyncTryGetChildrenNoThrow(ddl_zookeeper_path / task_path / "finished"));
+        ddl_task_status_paths.push_back(ddl_zookeeper_path / task_path / "active");
+        ddl_task_status_paths.push_back(ddl_zookeeper_path / task_path / "finished");
     }
 
+    auto ddl_tasks_info = zookeeper->get(ddl_task_full_paths);
```

Comment:
there is no simple interface for mixed type of requests so I opted for 2 sequential MultiReads.
Should be still okay compared to bunch of single requests.